Modules/_asynciomodule.c

```diff
@@ -341,8 +342,10 @@ get_running_loop(asyncio_state *state, PyObject **loop)
             }
         }
 
+        PyMutex_Lock(&state->mutex);
```

Comment:
The caching here is overly complicated, especially with a mutex, which will make it slower and possibly defeat the purpose of caching in the first place.

We should stick the running loop directly as a field in `PyThreadState` or `_PyThreadStateImpl` for fast access.

Reply:
I'll fix this in a follow-up PR. For now, I'll remove the locks.